src/devices/sound/mas3507d.h

```diff
@@ -47,7 +49,7 @@ class mas3507d_device : public device_t, public device_sound_interface
 	enum { IDLE, STARTED, NAK, ACK, ACK2 } i2c_bus_state;
 	enum { UNKNOWN, VALIDATED, WRONG } i2c_bus_address;
 
-	std::array<u8, 8000> mp3data;
+	std::array<u8, 0xe00> mp3data;
```

Comment:
I haven't been able to personally verify this myself, but someone familiar with this part of the hardware (who also was responsible for figuring out the original FPGA encryption) gave me the information that the game reads about 0xe00 bytes into the buffer to start with. This feels safer than 8000 bytes because reading in a large amount of data could potentially result in cases where you run ahead of the machine still loading data into memory, in the case of dynamically streamed audio.

Reply:
Ah nice, I had no way to get that information.  Would that someone when pkgingo?  Because if yes, I'm the one who gave him the decryption algorithm ;-)